rotkehlchen/chain/ethereum/decoding/decoder.py

```diff
@@ -289,17 +296,18 @@ def get_or_decode_transaction_events(
         events = self.decode_transaction(transaction, tx_receipt)
         return events
 
-    def _maybe_decode_simple_transactions(
-            self,
-            tx: EthereumTransaction,
-            tx_receipt: EthereumTxReceipt,
-    ) -> List[HistoryBaseEntry]:
-        """Decodes normal ETH transfers, internal transactions and gas cost payments"""
-        events: List[HistoryBaseEntry] = []
-        tx_hash_hex = tx.tx_hash.hex()
-        ts_ms = ts_sec_to_ms(tx.timestamp)
-
-        # check for internal transactions if the transaction is not canceled
+    def _maybe_decode_internal_transactions(
+        self,
+        tx: EthereumTransaction,
+        tx_receipt: EthereumTxReceipt,
+        events: List[HistoryBaseEntry],
+        tx_hash_hex: str,
+        ts_ms: TimestampMS,
+    ) -> None:
+        """
+        check for internal transactions if the transaction is not canceled. This function mutates
+        the events argument.
+        """
```

Comment:
you moved it to its own function. You can do better.

Exit early.

```python
if tx_receipt.status is False:
    return
```

Then rest of the function has better indentation.